File: Days 041 - 050/Day 45/Rand7.cs
using System;

namespace DailyCodingProblem
{
	internal class Day45
	{
		private static Random rng = new Random();

		private static int Main(string[] args)
		{
			string[] histogram = new string[7];

			for (int i = 0; i < 250; i++)
			{
				histogram[Rand7() - 1] += '*';
			}

			for (int i = 0; i < 7; i++)
			{
				Console.WriteLine($"{i + 1}: {histogram[i]}");
			}

			Console.ReadLine();

			return 0;
		}

		private static int Rand5() => rng.Next(1, 6);

		private static int Rand7()
		{
			int sum = 0;

			for (int i = 0; i < 7; i++)
			{
				sum += Rand5();
			}

			return sum % 7 + 1;
		}
	}
}

File: Days 041 - 050/Day 45/Rand7.cpp
#include <iostream>
#include <random>
#include <string>

unsigned int Rand5()
{
	static constexpr unsigned int MAX = 5;

	static std::random_device randomSeed;
	static std::mt19937 randomEngine(randomSeed());
	static const std::uniform_int_distribution<unsigned int> randomRange(1, MAX);

	return randomRange(randomEngine);
}

unsigned int Rand7()
{
	static constexpr unsigned int MAX = 7;

	unsigned int total = 0;

	for (unsigned int i = 0; i < MAX; i++)
	{
		total += Rand5();
	}

	total %= MAX;

	return ++total;
}

int main(int argc, const char* argv[])
{
	std::string histogram[7];

	for (unsigned int i = 0; i < 250; i++)
	{
		histogram[Rand7() - 1] += '*';
	}

	for (unsigned int i = 0; i < 7; i++)
	{
		std::cout << i + 1 << ": " << histogram[i] << "\n";
	}

	std::cin.get();

	return 0;
}